notes/transcript15.md

```diff
@@ -213,108 +213,228 @@ Igor: I guess the signature won't signatures won't match but we can do it a shot
+
+Jacek: Coming back a little bit to the user letting the user choose, I generally would consider that as fairly dangerous because or like from a security point of view tends to be a bad way to do it. One creative idea is to establish much longer time horizons on how fast these things change, so that the community has time to critique and review the change that's coming. 
+
+### 55:00 - 1:00:00
+
+Jacek: And then like it's a little bit similar if you look at the thing what we're doing there is basically did we call this the week subjectivity which is a way to deal with long range attacks and the working theory is here that if somebody does something obviously bad then over a long period of time, it's socially feasible to overcome this say years or a year, months whatever, like that but but a fairly long period where the bad behavior has time to establish itself and therefore would be known through outside means. If if we can. But what at least do basically is that the app needs to be able to support multiple protocols over extended period of time. And then the way to upgrade would well maybe we can hide it in an advanced option somewhere right but so that people can start trying these things and these new more efficient protocols for example but then also keep in mind when we develop things that that the overlap horizon or the overlap period should be considered to be fairly long and and and we should be able to provide you know a safer default for the user and then maybe switch that default and so on. And so as a way out from from from really bad decisions. I mean we do have the advantage that we're open and you know the code can be audited and so on, but then we also need this very long period of time for the information to seep out and sort of in case the code becomes tainted somehow, that there's time for this taint problem to become known even for people that are maybe not online all the time or more like you know that they know what kind of trend horizons we're looking at when we make these changes when they start using Dapps. So basically I know that alright if I start using status today, I know that for the sixth next six months, nothing's gonna change with the protocol and so on.
```

Comment:
```suggestion
Jacek: And then like it's a little bit similar if you look at the thing what we're doing there is basically did we call this the week subjectivity which is a way to deal with long range attacks and the working theory is here that if somebody does something obviously bad then over a long period of time, it's socially feasible to overcome this say years or a year, months whatever, like that but a fairly long period where the bad behavior has time to establish itself and therefore would be known through outside means. If we can. But what at least do basically is that the app needs to be able to support multiple protocols over extended period of time. And then the way to upgrade would well maybe we can hide it in an advanced option somewhere right but so that people can start trying these things and these new more efficient protocols for example, but then also keep in mind when we develop things that the overlap horizon or the overlap period should be considered to be fairly long and and and we should be able to provide you know a safer default for the user and then maybe switch that default and so on. And so as a way out from from from really bad decisions. I mean we do have the advantage that we're open and you know the code can be audited and so on, but then we also need this very long period of time for the information to seep out and sort of in case the code becomes tainted somehow, that there's time for this taint problem to become known even for people that are maybe not online all the time or more like you know that they know what kind of trend horizons we're looking at when we make these changes when they start using Dapps. So basically I know that alright if I start using status today, I know that for the six months, nothing's gonna change with the protocol and so on.
```